core/src/main/java/org/elasticsearch/cluster/ClusterModule.java

```diff
@@ -107,9 +107,24 @@ public ClusterModule(Settings settings, ClusterService clusterService, List<Clus
         this.clusterService = clusterService;
         this.indexNameExpressionResolver = new IndexNameExpressionResolver(settings);
         this.allocationService = new AllocationService(settings, allocationDeciders, shardsAllocator, clusterInfoService);
-        this.onStarted = () -> clusterPlugins.forEach(plugin -> plugin.onNodeStarted());
     }
 
+    public static Map<String, Supplier<ClusterState.Custom>> getCustomSuppliers(List<ClusterPlugin> clusterPlugins) {
```

Comment:
... and I agree the supplier approach here makes more sense because it will make rolling cluster restart handling  later easier.